bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_START);
+                            updateState(channel, currentStartState);
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_HOME);
+                            updateState(channel, currentHomeState);
+                        }
+                        break;
+                    case "JSON_BATTPERC":
+                        try {
+                            DecimalType battery = DecimalType.valueOf(value);
+                            if (!battery.equals(currentBattery)) {
+                                currentBattery = battery;
+                                channel = new ChannelUID(getThing().getUID(), CHANNEL_BATTERY);
+                                updateState(channel, battery);
+                            }
+                        } catch (NumberFormatException e) {
+                            logger.debug("Couldn't parse Battery Percent.");
+                            parsingOk = false;
+                        }
+                        break;
+                    case "CPU_IDLE":
+                        if (isLinked(CHANNEL_CPU_LOAD)) {
+                            try {
+                                DecimalType cpuLoad = new DecimalType(100 - Double.valueOf(value).longValue());
+                                if (!cpuLoad.equals(currentCPULoad)) {
+                                    currentCPULoad = cpuLoad;
+                                    channel = new ChannelUID(getThing().getUID(), CHANNEL_CPU_LOAD);
+                                    updateState(channel, cpuLoad);
+                                }
+                            } catch (NumberFormatException e) {
+                                logger.debug("Couldn't parse CPU Idle.");
+                                parsingOk = false;
+                            }
+                        }
+                        break;
+                    case "LGSRV_MEMUSAGE":
+                        if (!value.equals(currentSrvMem)) {
+                            currentSrvMem = value;
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_SRV_MEM);
+                            updateState(channel, StringType.valueOf(value));
+                        }
+                        break;
+                    case "JSON_TURBO":
+                        OnOffType turbo = value.equalsIgnoreCase("true") ? OnOffType.ON : OnOffType.OFF;
+                        if (!turbo.equals(currentTurbo)) {
+                            currentTurbo = turbo;
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_TURBO);
+                            updateState(channel, turbo);
+                        }
+                        break;
+                    case "JSON_REPEAT":
+                        OnOffType repeat = value.equalsIgnoreCase("true") ? OnOffType.ON : OnOffType.OFF;
```

Comment:
I think this could/should just be:

```java
currentRepeat = OnOffType.from("true".equalsIgnoreCase(value));
updateState(CHANNEL_REPEAT, currentRepeat);
```

This also applies to most other cases.